internal/cloud/gcp/compute.go

```diff
@@ -265,17 +265,30 @@ func (g *GCP) ComputeImageShare(ctx context.Context, imageName string, shareWith
 //
 // Uses:
 //	- Compute Engine API
-func (g *GCP) ComputeImageDelete(ctx context.Context, image string) error {
+func (g *GCP) ComputeImageDelete(ctx context.Context, resourceId string) error {
 	computeService, err := compute.NewService(ctx, option.WithCredentials(g.creds))
 	if err != nil {
 		return fmt.Errorf("failed to get Compute Engine client: %v", err)
 	}
 
-	_, err = computeService.Images.Delete(g.creds.ProjectID, image).Context(ctx).Do()
+	_, err = computeService.Images.Delete(g.creds.ProjectID, resourceId).Context(ctx).Do()
 
 	return err
 }
 
+func (g *GCP) ComputeExecuteForImage(ctx context.Context, f func(*compute.ImageList) error) error {
```

Comment:
Nitpick: I find the method name confusing - if I understand ti correctly, it filters the image list using the provided function, but that is not what I would expect based on its name. But maybe the problem is in me 😄

Reply:
It executes `f` with a list of all the images in compute.

Maybe like, `ComputeExecuteFunctionForEachImage`? Make it a bit more explicit? Happy to take other suggestions of course.